lib/extension_scanner.rb

```diff
@@ -147,8 +147,12 @@ def scan(filter)
     extensions.each do |e|
       queue_requests(e['name']) do |resp, extension_path, manifest_uri|
         lock.synchronize do
-          res = process_result(e, extension_path, manifest_uri, resp.body)
-          detected.push(res) if res[:vulns].length > 0
+          begin
+            res = process_result(e, extension_path, manifest_uri, resp.body)
```

Comment:
This is pretty a pretty broad rescue block. It will swallow up other errors that may come up. You should put the rescue block right where the error occurs (around the version parse). If version can't be parsed then don't continue. Make sure your code that parses the version out (and passes it to the Gem::Version) is correctly grabbing the version. If it is, and this is some edge case, then the rescue block will be fine.